python/src/gluon_ir.cc

```diff
@@ -355,15 +357,19 @@ void init_gluon_ir(py::module &&m) {
            })
       .def("get_padded_shared_layout",
            [](GluonOpBuilder &self, std::vector<unsigned> &intervals,
```

Comment:
Do you want to kill the previous builder? Isn't it useful?

Reply:
Maybe I am wrong here, but having two doesn't make sense because we call them in `_to_ir` inside the python class, so the gluon kernel should not use those builder functions directly?

Instead I added a static function to the python class `with_identity_for` which offers a shortcut to create the padded layout with an identity mapping.

Reply:
Ah, I had missed that as I hadn't finished revieweing the rest. Will review the whole thing on Monday